app/javascript/components/widgets/numbers/sleeping-blocks.jsx

```diff
@@ -55,13 +55,18 @@ Sleeping.update = function freezeOnSleep(bodies, timeScale) {
       ) {
         body.sleepCounter += 1;
 
-        if (body.sleepCounter >= body.sleepThreshold) {
+        if (body.sleepCounter >= body.sleepThreshold && body.render.visible) {
           // Bodies here set to static instead of being put to sleep
           Body.setStatic(body, true);
+          body.render.visible = false;
+          addToOverlay(body);
```

Comment:
This function is doing a little more than just `freezeOnSleep` now. Its also hiding the bodies and calling a callback. I think function composition may be a useful pattern here.